aqt/metadata.py

```diff
@@ -351,6 +353,65 @@ def possible_extensions_for_arch(arch: str) -> List[str]:
             return [ext_lt_6]
         return [ext_lt_6, ext_ge_6]
 
+    # Architecture, as reported in Updates.xml
+    MINGW_ARCH_PATTERN = re.compile(r"^win(?P<bits>\d+)_mingw(?P<version>\d+)?$")
+    # Directory that corresponds to an architecture
+    MINGW_DIR_PATTERN = re.compile(r"^mingw(?P<version>\d+)?_(?P<bits>\d+)$")
+
+    @staticmethod
+    def select_default_mingw(mingw_arches: List[str], is_dir: bool) -> Optional[str]:
+        """
+        Selects a default architecture from a non-empty list of mingw architectures, matching the pattern
+        MetadataFactory.MINGW_ARCH_PATTERN. Meant to be called on a list of installed mingw architectures,
+        or a list of architectures available for installation.
+        """
+
+        ArchBitsVer = Tuple[str, int, Optional[int]]
+        pattern = QtRepoProperty.MINGW_DIR_PATTERN if is_dir else QtRepoProperty.MINGW_ARCH_PATTERN
+
+        def mingw_arch_with_bits_and_version(arch: str) -> Optional[ArchBitsVer]:
+            match = pattern.match(arch)
+            if not match:
+                return None
+            bits = int(match.group("bits"))
+            ver = None if not match.group("version") else int(match.group("version"))
+            return arch, bits, ver
+
+        def select_superior_arch(lhs: ArchBitsVer, rhs: ArchBitsVer) -> ArchBitsVer:
+            _, l_bits, l_ver = lhs
+            _, r_bits, r_ver = rhs
+            if l_bits != r_bits:
+                return lhs if l_bits > r_bits else rhs
+            elif r_ver is None:
+                return lhs
+            elif l_ver is None:
+                return rhs
+            return lhs if l_ver > r_ver else rhs
+
+        candidates: List[ArchBitsVer] = list(filter(None, map(mingw_arch_with_bits_and_version, mingw_arches)))
+        if len(candidates) == 0:
+            return None
+        default_arch, _, _ = reduce(select_superior_arch, candidates)
+        return default_arch
+
+    @staticmethod
+    def find_installed_qt_mingw_dir(installed_qt_version_dir: Path) -> Optional[Path]:
+        """
+        Locates the default installed qt mingw directory.
+
+        :param installed_qt_version_dir: A directory that may contain a qt-mingw installation.
+                                         It should look something like `.../Qt/6.3.0/`, and contain qt installations.
+        """
+
+        def contains_qmake_exe(arch_path: Path) -> bool:
+            return (arch_path / "bin/qmake.exe").is_file()
+
+        paths = [d for d in installed_qt_version_dir.glob("mingw*")]
+        directories = list(filter(contains_qmake_exe, paths))
+        arches = [d.name for d in directories]
+        selected = QtRepoProperty.select_default_mingw(arches, is_dir=True)
+        return installed_qt_version_dir / selected if selected else None
+
```

Comment:
`QtRepoProperty` is now 100 lines, and it's all static methods. It doesn't really have anything to do with the MetadataFactory. Maybe it should be a module instead of a class, and exist in its own file?